app/code/Magento/Inventory/Model/Indexer/StockItem/Action/Row.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\Indexer\StockItem\Action;
+
+use Magento\Inventory\Model\Indexer\StockItem\AbstractAction;
+
+class Row extends AbstractAction
+{
+    /**
+     * Execute Row reindex
+     *
+     * @param int|null $id
+     * @throws \Magento\Framework\Exception\LocalizedException
+     *
+     * @return void
+     */
+    public function execute($id = null)
+    {
+        if (!isset($id) || null === $id) {
+            throw new \Magento\Framework\Exception\LocalizedException(
+                __('We can\'t rebuild the index for an undefined stock item .')
+            );
+        }
+        try {
+            $this->reindexRows([$id]);
+        } catch (\Exception $e) {
+            throw new \Magento\Framework\Exception\LocalizedException(__($e->getMessage()), $e);
```

Comment:
it's not correct translation,
`__($e->getMessage()), $e);`
Because \Exception $e ,so it's not a successor of Localized Exception, thus $e->getMessage()
could return a string for which you don't have precise translation